coreclient/src/groups/process.rs

```diff
@@ -133,38 +130,17 @@ impl Group {
                     InfraAadPayload::GroupOperation(group_operation_payload) => {
                         // Process adds if there are any.
                         if !group_operation_payload
-                            .new_encrypted_identity_link_keys
+                            .new_encrypted_user_profile_keys
                             .is_empty()
                         {
-                            let number_of_adds = staged_commit.add_proposals().count();
-                            let number_of_ilks = group_operation_payload
-                                .new_encrypted_identity_link_keys
-                                .len();
-                            let number_of_upks = group_operation_payload
-                                .new_encrypted_user_profile_keys
-                                .len();
-                            // Make sure the vector lengths match.
-                            let vector_lengths_match = number_of_adds == number_of_ilks
-                                && number_of_adds == number_of_upks;
-                            ensure!(
-                                vector_lengths_match,
```

Comment:
Since we're zipping them later, we should still `ensure!()` that the number of Adds and the number of new encrypted use profile keys are the same.

I know this is slightly out of scope, but now that I think about it, we should test this even before the `if`, i.e. we should make sure that if there are no new encrypted user profile keys there also no Adds in the commit.

Reply:
Good catch. Fixed in 9e9612c